web/src/app/monitors/monitor-create-edit/monitor-create-edit.component.ts

```diff
@@ -0,0 +1,91 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { v4 as uuid } from 'uuid';
+
+// Third party modules
+import { ActivatedRoute } from '@angular/router';
+import { Subscription } from 'rxjs';
+
+// Dashboard hub models and services
+import { MonitorService, ProjectService } from '../../core/services/index.service';
+import { MonitorModel, ProjectModel } from '../../shared/models/index.model';
+
+@Component({
+  selector: 'dashboard-monitor-create-edit',
+  templateUrl: './monitor-create-edit.component.html',
+  styleUrls: ['./monitor-create-edit.component.scss'],
+})
+export class MonitorCreateEditComponent implements OnInit {
+
+  private id: string;
+  private monitorsList: MonitorModel[] = [];
+  private projectSubscription: Subscription;
+  private uid: string;
+  public isEdit: Boolean = false;
+  public monitorForm: FormGroup;
+
+  constructor(
+    private form: FormBuilder,
+    private monitorService: MonitorService,
+    private projectService: ProjectService,
+    private route: ActivatedRoute
+  ) { }
+
+  /**
+   * Lifecycle init method
+   */
+  ngOnInit(): void {
+    this.uid = this.route.snapshot.paramMap.get('uid');
+    this.intializeMonitorForm();
+    this.projectSubscription = this.projectService
+      .findOneById(this.uid)
+      .subscribe((project: ProjectModel) => {
+        this.monitorsList = project.monitors ? project.monitors : [];
+        this.id = this.route.snapshot.paramMap.get('id');
+        if (this.id) {
+          this.isEdit = true;
+          const filteredMonitor: MonitorModel = this.monitorService.findMonitorById(this.monitorsList, this.id);
```

Comment:
I think this would be better on a `resolver`, sometime I get an empty form then data loads

Reply:
Resolver is required when we need to check some access on path if we have to navigate or not. 
Please let me know the scenario when you get the empty form and then data loads ?

Reply:
To check access first would be a `guard`, a `resolver` is to load the data before the page loads. In this case, we can load the data before the form is displayed.

The form is sometimes empty with a slow connection on the 2nd visit, then it is populate, a resolve would fix this issue as the form won't display until the data is ready

Reply:
done